### PR TITLE
Adds a 'clear search' icon, solves issue #360

### PR DESCRIPTION
**Pull request by schester44**

### CODE REVIEW

File: src/renderer/components/main/middlebar/toolbar_search.tsx [hunk 1]
```diff
@@ -37,9 +54,10 @@ class Search extends React.Component<any, undefined> {
 
 export default connect ({
   container: Main,
-  shouldComponentUpdate: false,
+  shouldComponentUpdate: true,
```
Comment (schester44) [inline src/renderer/components/main/middlebar/toolbar_search.tsx]: I'm thinking this isn't entirely correct but I wasn't sure what the correct value should be.

**Comment by fabiospampinato** [comment]
The implementation is too verbose though and the icon isn't really displayed like a button.

But I like the idea a lot!

**Comment by RomanKornev** [comment]
Closes #360

File: src/renderer/components/main/middlebar/toolbar_search.tsx [hunk 2]
```diff
@@ -9,22 +9,39 @@ import Main from '@renderer/containers/main';
...
+    this.ref.current.value = ""
+  }
+
   render () {
+    const isSearching = this.props.query && this.props.query.length > 0
 
     return (
       <div className="multiple joined no-separators grow search">
         <input ref={this.ref} autoFocus type="search" className="bordered grow small" placeholder="Search notes..." defaultValue={this.props.query} onChange={this.onChange} />
         <div className="label bordered xsmall" title="Search">
```
Comment (RomanKornev) [inline src/renderer/components/main/middlebar/toolbar_search.tsx]: Tooltip should also update.

**Comment by schester44** [comment]
@fabiospampinato Any suggestions for making it less verbose?

I'm able to remove the constructor if I bind `this` to the onClearSearch. Also able to remove the duplicate `<i>` icon elements by doing something like:

`<i className="icon" onClick={this.onClearSearch.bind( this )}>{isSearching ? 'close' : 'magnify'}</i>`

We could also remove the onClearSearch all together if we had an always-accurate search term value but I don't think thats possible due to delay while debouncing.

As for the icon not acting like a button, did you have something in mind for how this would look/function? I like the current UI implementation but agree it doesn't display/act like a button. UX wise, an opacity change on hover would help but ultimately the execution is your call.

**Comment by fabiospampinato** [comment]
@schester44 I'd like to merge this right way, could you accept the CLA?

**Comment by schester44** [comment]
@fabiospampinato Accepted on my end 👍

**Comment by fabiospampinato** [comment]
@schester44 I'm not sure why it says that it's still pending 🤔 Did you click on the "sign in with GitHub and accept" button?

**Comment by schester44** [comment]
@fabiospampinato I was wondering that myself. I did click the sign in button.

This is what i'm seeing:
https://i.imgur.com/KFzDyhK.png

**Comment by fabiospampinato** [comment]
@schester44 weird, let's consider it accepted then. 

Thanks for the contribution, next time don't commit your yarn.lock file though 👍

**Pull request merged**